File: app/src/test/java/com/example/solar_alarm/HttpTestCase.java
package com.example.solar_alarm;

import org.junit.Assert;
import org.junit.Test;

import java.io.IOException;
import java.util.Calendar;

public class HttpTestCase
{
    protected Calendar eventDate;

    @Test
    public void HttpGetTest()
    {
        SunriseSunsetRequest sunriseSunsetRequest = new SunriseSunsetRequest((float) 40.67441, (float) -73.43162, Calendar.getInstance());

        try
        {
            HttpRequests httpRequests = new HttpRequests(sunriseSunsetRequest);
            SunriseSunsetResponse response = httpRequests.GetSolarData(sunriseSunsetRequest);

            Assert.assertSame("OK", response.status);
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }
    }
}

File: app/src/test/java/com/example/solar_alarm/HttpTestCase.kt
package com.example.solar_alarm

import com.example.solar_alarm.sunrise_sunset_http.HttpRequests
import com.example.solar_alarm.sunrise_sunset_http.SunriseSunsetRequest
import com.example.solar_alarm.sunrise_sunset_http.SunriseSunsetResponse
import kotlinx.coroutines.runBlocking
import org.junit.Assert
import org.junit.Test
import java.io.IOException
import java.time.LocalDate

class HttpTestCase
{
    @Test
    fun HttpGetTest()
    {
        try
        {
            val httpRequests          = HttpRequests()
            val sunriseSunsetRequest  = SunriseSunsetRequest(40.67441.toFloat(), (-73.43162).toFloat(), LocalDate.now())
            val sunriseSunsetResponse : SunriseSunsetResponse

            runBlocking {
                sunriseSunsetResponse = httpRequests.GetSolarData(sunriseSunsetRequest)
            }

            Assert.assertNotNull(sunriseSunsetResponse)
            Assert.assertSame("OK", sunriseSunsetResponse!!.status)
        }
        catch (e: IOException)
        {
            e.printStackTrace()
        }
    }
}
